### PR TITLE
Allow for code insertion and modification by exercises

### PR DESCRIPTION
**Pull request by AydinHassan**
* Allows exercises to insert code before or after submission
* Allows exercises to completely modify submissions, eg wrap in an exception catch block
* Provide some default insertions for date timezone and display errors

Solves #5

### CODE REVIEW

**Comment by AydinHassan** [comment]
@mikeymike check this bad boy

File: app/config.php [hunk 1]
```diff
@@ -14,11 +15,16 @@
 use PhpParser\Parser;
 use PhpParser\ParserFactory;
 use PhpSchool\PhpWorkshop\Check\CgiOutputCheck;
+use PhpSchool\PhpWorkshop\Check\CodeParseCheck;
+use PhpSchool\PhpWorkshop\CodeModification as M;
```
Comment (mikeymike) [inline app/config.php]: `M` ... just a thought maybe `Mod` or something would be a little better.

**Comment by mikeymike** [comment]
NICE WORK DUDE!!!! 

Seriously this looks sweet, pulled it down locally and had a mess around with the tests too and it seems pretty solid!

File: app/config.php (hunk 1 above)
Comment (AydinHassan) [inline app/config.php]: Yeah, i'm gonna rename it to `CodeInsertion` anyway, it doesn't really modify code

**Pull request merged**